nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,297 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.epgstation;
+
+  username = config.users.users.epgstation.name;
+  groupname = config.users.users.epgstation.group;
+
+  settingsFmt = pkgs.formats.json {};
+  settingsTemplate = settingsFmt.generate "config.json" cfg.settings;
+  preStartScript = let
+    readPassword = f: if (f != null)
+      then "$(head -n1 \"${f}\")"
+      else "INSECURE_GO_CHECK_CONFIGURATION_NIX";
+  in
+  pkgs.writeScript "epgstation-prestart" ''
+    #!${pkgs.runtimeShell}
+
+    PASSWORD="${readPassword cfg.basicAuth.passwordFile}"
+    DB_PASSWORD="${readPassword cfg.database.passwordFile}"
+
+    # setup configuration
+    sed \
+      -e "s,@password@,$PASSWORD,g" \
+      -e "s,@dbPassword@,$DB_PASSWORD,g" \
+      ${settingsTemplate} > /etc/epgstation/config.json
+    chmod 440 /etc/epgstation/config.json
+    chown "${username}:${groupname}" /etc/epgstation/config.json
```

Comment:
This is a theoretical race condition, maybe the permissions and ownership should be set first.

Reply:
I'm not sure I follow. Are you worried that Epgstation would attempt to access the configuration file before it's ready? If so, I've specified systemd to run this script as root before Epgstation starts.

Reply:
I almost commented on this asking to change it back - mind adding a comment explaining why we're doing this?

Reply:
The `chown` is still after the `sed`. It shouldn't make much difference though, since the default owner is probably root. Still, it would be slightly nicer to set the proper owner first.